src/write.c

```diff
@@ -908,31 +910,27 @@ static avifResult avifEncoderAddImageItems(avifEncoder * encoder,
         gridItem->gridRows = gridRows;
         gridItem->gridWidth = gridWidth;
         gridItem->gridHeight = gridHeight;
-        *mainItemID = gridItem->id;
+        *topLevelItemID = gridItem->id;
     }
 
     for (uint32_t cellIndex = 0; cellIndex < cellCount; ++cellIndex) {
         avifEncoderItem * item = avifEncoderDataCreateItem(encoder->data, "av01", infeName, infeNameSize, cellIndex);
+        AVIF_CHECKERR(item, AVIF_RESULT_OUT_OF_MEMORY);
```

Comment:
Nit/optional: For something so simple, we don't need to use a macro:

```
        if (!item) {
            return AVIF_RESULT_OUT_OF_MEMORY;
        }
```

Reply:
Do you prefer not to use macros? To me they provide the following benefits:
- Shorter code, easier to read
- You can put a breakpoint into it to speed up debugging, if used everywhere

At the expense of indirection, but I believe most people are used to this kind of check-ok-or-return pattern nowadays.

Reply:
Thank you. I checked the src/read.c file and found several similar uses of this macro.